analysis_driver/dataset.py

```diff
@@ -96,13 +101,13 @@ def _is_ready(self):
         raise NotImplementedError
 
     def __str__(self):
-        out = [self.name]
+        s = self.name
         pid = self.most_recent_proc.get('pid')
         if pid:
-            out.append('(%s)' % pid)
+            s += ' (%s)' % pid
         if self.stages:
-            out.append('-- %s' % (', '.join(self.stages)))
-        return ' '.join(out)
+            s += ' -- ' + ', '.join(self.stages)
+        return s
```

Comment:
We should cache ```self.stages``` here to avoid querying the REST API twice
```python
stage = self.stages
if stages:
    s += ' -- ' + ', '.join(stages)
```